.github/workflows/rust.yml

```diff
@@ -5,21 +5,91 @@ on:
     branches: ["main"]
   pull_request:
     branches: ["main"]
+  workflow_dispatch:  # Allow manual triggering
 
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  build:
+  lint:
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v4
       - name: Run cargo-fmt
         run: cargo fmt --all -- --check
       - name: Run cargo-clippy
         run: cargo clippy -- -D warnings
-      - name: Build
-        run: cargo build --verbose
+      - name: Check
```

Comment:
I broke the jobs into multiple ones that ran in parallel once `clippy` and `fmt` and `check` pass so avoid wasting I runner time on jobs that will not compile